src/app/core/services/label.service.ts

```diff
@@ -69,6 +72,91 @@ export class LabelService {
     return color;
   }
 
+  /**
+   * Verifies that each label of its corresponding type exists in the current
+   * repository and creates one if it does not exist.
+   */
+  private createMissingLabels(): void {
+    for (const labelType of Object.keys(ISSUE_LABELS)) {
+      const compulsoryLabels: string[] = ISSUE_LABELS[labelType];
+
+      compulsoryLabels.forEach(labelName => {
+        if (this.labelExists(labelType, labelName)) {
+          return;
+        }
+
+        const color: string = this.getRandomLabelColour();
+
+        // Update Local Data.
+        this.getLabelList(labelType).push({labelValue: labelName, labelColor: color});
+        this.labelColorMap.set(labelName, color);
+
+        // Update Remote Data.
+        this.githubService.createLabel(this.getFormattedLabelName(labelType, labelName), color);
+      });
+    }
+  }
+
+  /**
+   * Verifies that every label has the correct color and updates a label's color
+   * if otherwise.
+   */
+  private validateLabelColors(): void {
+    for (const labelType of Object.keys(ISSUE_LABELS)) {
+
+      this.getLabelList(labelType).forEach(label => {
+        if (this.labelHasCorrectColor(labelType, label.labelValue)) {
+          return;
+        }
+
+        const color: string = this.getRandomLabelColour();
+
+        // Update Local Data
+        label.labelColor = color;
+        this.labelColorMap.set(label.labelValue, label.labelColor);
+
+        // Update Remote Data.
+        this.githubService.updateLabel(this.getFormattedLabelName(labelType, label.labelValue), color);
+      });
+    }
+  }
```

Comment:
hmm... this doesn't look right. how about this?
* First, there should be a place where we define the _expected_ labels and their correct color (yes, the colors should be fixed by us). Does such a list exist in the code already?
* Next, there should be a function that takes a list of actual labels and a list of expected labels; iterates through the expected labels; and updates the repo for each missing/mis-matching label. i.e., as much as possible functions should be stateless -- all they need are given via parameters instead of accessing data directly somewhere in the body of the function.
Would that work?

Reply:
@damithc 
Would this color scheme be okay for the labels?
<img src="https://user-images.githubusercontent.com/39243082/59736988-75bf6680-928e-11e9-947f-94acc6b7f9a5.png" width="300px"/>

Reply:
FunctionalityBug can be darker shade of the same color. Similarly, Incomplete can be darker. The rest is OK. Define them in one place so that we can tweak later.

Reply:
> * Next, there should be a function that takes a list of actual labels and a list of expected labels; iterates through the expected labels; and updates the repo for each missing/mis-matching label. i.e., as much as possible functions should be stateless -- all they need are given via parameters instead of accessing data directly somewhere in the body of the function.
>   Would that work?

Yes I have refactored the process.

Reply:
> FunctionalityBug can be darker shade of the same color. Similarly, Incomplete can be darker. The rest is OK. Define them in one place so that we can tweak later.

Sure! Label Colors are stored in [`label.model.ts`](https://github.com/ptvrajsk/CATcher/blob/LabelSetup/src/app/core/models/label.model.ts)

Reply:
@damithc 
Is this better?
<img src="https://user-images.githubusercontent.com/39243082/59739755-78bf5480-9298-11e9-9010-6a4bf3b751a0.png" width="200px"/>

Reply:
Yes, let's go with that